internal/pkg/etcd/etcd.go

```diff
@@ -148,53 +149,75 @@ func validateMemberHealth(ctx context.Context, memberURIs []string) (err error)
 }
 
 // LeaveCluster removes the current member from the etcd cluster and nukes etcd data directory.
-func (c *Client) LeaveCluster(ctx context.Context) error {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return err
-	}
-
-	if err = c.RemoveMember(ctx, hostname); err != nil {
+func (c *Client) LeaveCluster(ctx context.Context, memberID string) error {
```

Comment:
if this method goes that far as stopping the services, it might find local member ID itself? otherwise someone might call it instead of the 'RemoveMember'